core/common/src/MapLayerSettings.ts

```diff
@@ -19,6 +19,10 @@ export type ImageryMapLayerFormatId  = "ArcGIS" | "BingMaps" | "MapboxImagery" |
 /** @public */
 export type SubLayerId = string | number;
 
+/** @public */
+export type  PropertyBagArrayProperty = Array<number|string|boolean>;
+export interface PropertyBag { [key: string]: number | string | boolean | PropertyBagArrayProperty };
```

Comment:
See if you can avoid reinventing [JSONSchema](https://github.com/iTwin/itwinjs-core/blob/master/core/bentley/src/JsonSchema.ts).

Reply:
`export type JSONSchemaType = string | number | boolean | object | JSONSchemaType[];`

I need a type representing types that can be directly serialized and can be compared easily.  This is why I excluded `object`, and I didn't want to support array of arrays.

Is there a way to compose type from existing types with typescript ?

Reply:
Do the google maps property bags definitely conform to those constraints? Do we anticipate other map providers that might not meet those constraints?

Reply:
Currently for google it only holds this data structure:
```
export interface CreateGoogleMapsSessionOptions {
  mapType: GoogleMapsMapType;
  language: string;     
  region: string;    
  orientation?: number;
  layerTypes?: string[];
};
```
My thinking is that MapLayerSettings's properties are meant to be simple list key,value pairs that can be passed down to the provider, it's not meant to pass complex data structures.  If nested objects need to be passed, it can always be simplified into a flat list of values and re-construct inside the provider.

Reply:
Oh, so this property bag is not something supplied by the maps service, it's a way for the user of our API to configure specific map providers?

Reply:
Exactly.. sorry for the confusion, I should have better explained this.

Reply:
Thanks. I'd suggest using less generic names than "PropertyBag" and "PrimitiveType" (especially since EC already has *two* different "primitive type" types).

Reply:
I renamed `PropertyBag` to `MapLayerProviderProperties`.  If you have better terminology, let me know.

 Also, I now refer `number|string|boolean` as `SimpleType`. I'm unsure if it deserve a type definition in `@itwin/core-bentley` though.